### ISSUE TITLE
Sending an email caches a view with the wrong group (ownership)

### ISSUE THREAD

**Issue by Hiroki111** [laravel/framework#28356]
- Laravel Version: 5.7
- PHP Version: 7.1.3
- Database Driver & Version: MySQL / 5.6.43

### Description:
When my app sends an email with a `Job`, it caches the view of the email.
However, the ownership of the cached files is incorrect.
When I run `ls -al storage/framework/views`, I can see that each file looks something like...

`-rw-rw-r--  1 user user   192 Apr 27 09:46 4251.....php`

This app is running on Nginx, and all the files in storage (and bootstrap) has its owner user and the group nginx.

i.e.
`drwxrwxr-x.   5 user nginx   46 Feb 23 02:50 storage`
If there is a file the group of which isn't nginx, there will be the "Permission denied" error.

Therefore, each cached file should have its ownership be like...

`-rw-rw-r--  1 user nginx   192 Apr 27 09:46 4251.....php`

If I change the ownership of cached views with `chown $USER:nginx -R storage/`, the "Permission denied" error won't be caused. **_This is why I assume that this is a bug_**.

### Steps To Reproduce:

First of all, I'm using `CentOS 7`, which (unlike Ubuntu) isn't using `www-data` user/group.

To reproduce the bug, it'll be necessary to create a job with your DB.
In other words, run the following commands.
`php artisan queue:table`
`php artisan migrate`

Then, send an email from a controller with a job in the following way.

    class ExampleController extends Controller
    {     
        // __construct

        public function store()
        {
            //other lines here...
            SendEmail::dispatch($order);
        }
    }

    class SendEmail implements ShouldQueue
    {
        use Dispatchable, InteractsWithQueue, Queueable, SerializesModels;

        //__construct

        public function handle()
        {
            Mail::to($this->order->email)->send(new OrderEmail($this->order));
        }
    }

**Comment by mfn** [comment]
Are you running your webserver and your workers under different users?

**Comment by Hiroki111** [comment]
My web server (Nginx) is running under `nginx`.

i.e.
When I ran `ps aux|grep nginx`, I can see....

    root      6456  0.0  0.0 126504  5468 ?        Ss   Apr26   0:00 nginx: master process /usr/sbin/nginx
    nginx     6509  0.0  0.2 719604 37588 ?        S    Apr26   0:01 php-fpm: pool www
    //simlar lines here
    user    8268  0.0  0.0 112708   980 pts/0    S+   08:53   0:00 grep --color=auto nginx
    nginx    12138  0.0  0.0 138756  4444 ?        S    Apr27   0:00 nginx: worker process
    //again, other similar lines
    nginx    18688  0.0  0.2 717452 35300 ?        S    Apr26   0:01 php-fpm: pool www

As far as I understand from these lines, the webserver is run by the `nginx` user.

However, I'm not sure what you mean by "your workers".
In the snippet above, I can see "`nginx: worker process`". 
Is this what you're asking about?

**Comment by sisve** [comment]
The job is executed by a queue worker. It isn't visible in the ps data you've sent us. It's not the same as "nginx: worker process". Try `ps aux | grep queue` and you'll probably find something executing "php artisan queue:work" or "php artisan queue:listen". That process seems to be running with another user account than what nginx is using.

Why do you two separate accounts; a user account, and a nginx account? Configure the fpm pool to use your user account, and the php code will execute as your user, have the same permissions as your user, and be able to do the exact same things as your user.

**Comment by Hiroki111** [comment]
Thank you for your reply.

    user   32212  0.0  0.0 112708   980 pts/0    S+   09:59   0:00 grep --color=auto queue

This is the result of `ps aux | grep queue`, and I can see that the queue is owned by `user`.

"Why do you two separate accounts; a user account, and a nginx account? "

The reason why I was using `nginx` account was because the server, Nginx, was run by the `nginx` user account. The server had be able to access the Laravel app.
So, I changed the ownership status for files in `storage` (and `bootstrap`), so that the user is `$USER` and the group is `nginx`

but now _**I've changed the user of Nginx into `user`**_, so that Nginx can access `storage/framework/view`
I'll see if this is the right approach.

**Comment by sisve** [comment]
There's two different components here.

1. nginx handles incoming http requests and tries to figure out what to do with it. It may need readonly access to source files to determine if they exist at all.
2. php-fpm handles actually executing the code. This needs readonly access to the code, and write-access to places you need to write.

You should probably change the user of the php-fpm pool, not nginx.

**Comment by Hiroki111** [comment]
@sisve 
I understand your explanations on Nginx and php-fpm, but I'm not really sure why php-fpm's user should be changed (instead of Nginx's user).

On Ubuntu, I used to have Nginx and Laravel files' owned by `www-data`.
(When I google something like "setup Nginx on Ubuntu for Laravel", I see many examples which use `www-data` as the owner of Nginx and Laravel files, like [this post](https://websiteforstudents.com/install-laravel-on-ubuntu-17-04-17-10-with-nginx-mariadb-and-php-support/) and [this post](https://medium.com/@asked_io/how-to-install-php-7-2-x-nginx-1-10-x-laravel-5-6-f9e30ee30eff).

This means, in my opinion, that Nginx and Laravel apps are supposed to be owned by the same user, not php-fpm.

Sorry this is a bit off-topic, but could you elaborate on why it is php-fpm that should change its user?

**Comment by sisve** [comment]
The process that executes the code needs to be using the proper user account. It's the fpm that executes the code, so by that logic it's fpm that should be using the user account.

Keep in mind that the fpm supports several pools, and you can setup on for every user account. That way you can have several different php applications executing with different php permissions. However, you can only have one nginx process running.

I believe that nginx doesn't even need permissions to read the files if it is configured to send everything to the fpm. That way nginx will blindly accept anything, and forward it to the fpm, which is executing as the proper user and doing the actual work. This setup will however have downsides with static files which is no longer optimized in nginx but served via the fpm.

**Comment by driesvints** [comment]
Going to close this because we don't offer support for this version anymore. Please try to ask this on a support channel next time (see the issue templates).
